streampipes-client-python/streampipes_client/functions/broker/nats_broker.py

```diff
@@ -0,0 +1,81 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+from typing import AsyncIterator
+
+from nats import connect
+from nats.aio.client import Client
+from nats.aio.subscription import Subscription
+from streampipes_client.functions.broker.broker import Broker
+
+logger = logging.getLogger(__name__)
+
+
+class NatsBroker(Broker):
+    """Implementation of the NatsBroker"""
+
+    nats_client: Client
+    subscription: Subscription
+
+    async def _makeConnection(self, host_address: str, port: int) -> None:
+        """Helper function to connect to a server.
+
+        Parameters
+        ----------
+
+        host_address: str
+            The host address of the server, which the broker connects to.
+
+        port: int
+            The port number of the connection.
+
+        Returns
+        -------
+        None
+        """
+        self.nats_client = await connect(f"nats://{host_address}:{port}")
+        if self.nats_client.connected_url is not None:
```

Comment:
else should raise an exception here

Reply:
I don't really know if it's even possible that the else case happens because I think their should always be a `connected_url` if the connection was successful. I only added the if because the type checking failed as the url could be `None`.